### PR TITLE
Banners (beta + fetch errors)

### PR DESCRIPTION
**Pull request by sidneymbell**
image
image

### CODE REVIEW

File: src/routes/fetchTree.tsx [hunk 1]
```diff
@@ -31,11 +31,7 @@ async function handleDataFetch(targetUrl: string, dispatch: Function) {
   try {
     response = await axios.get(targetUrl);
   } catch {
-    const errorMessage =
-      `Attempted to fetch data from URL: ${targetUrl}, but ` +
-      "there was a problem with fetching the data. Please confirm " +
-      "the URL is publicly accessible and has the appropriate CORS policy " +
-      "in place. You may want to look in your browser console.";
+    const errorMessage = `We weren't able to import your tree data. Please confirm this URL is correct and publicly accessible: ${targetUrl}`;
```
Comment (vincent-czi) [inline src/routes/fetchTree.tsx:34]: This is more human-readable, but I do kind of want the bread crumbs about CORS... I foresee that being an issue that might get encountered and can be really flummoxing if you have never seen a problem like that before. Although I guess their browser console would at least mention CORS. I suppose the real thing I want is for this to flag to the user, "hey, something went wrong in a way you probably have the power to fix, but will require a reasonable amount of technical know-how." Being able to take the `targetUrl` value, paste it in to a browser, and get JSON is _not enough_ to guarantee Galago can get it, and that's by the design of how browsers work.

I have no idea how to unpack all of that meaning into a short error message, so maybe this approach is the way to go, but that's what I was thinking.

File: src/reducers/global.ts [hunk 2]
```diff
@@ -362,6 +362,16 @@ export const global = (state = defaultState, action: any) => {
       };
     }
 
+    case ACTION_TYPES.FETCH_ERROR_CLEAR: {
```
Comment (vincent-czi) [inline src/reducers/global.ts:365]: Nit: This is not technically an `ERROR_CLEAR`, it's a clear of the display, more `ERROR_DISPLAY_CLEAR`. The prior `errorMessage` and `errorDuringFetch` values will remain after this reducer happens. I think that's a fine design choice -- you might want to clear the error warning being displayed but still have some part of your app know the prior fetch failed -- but the `FETCH_ERROR_CLEAR` to feels like it implies a full clear of everything about error.

File: src/components/header/fetchError.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        severity="error"
+        style={{
+          width: 450,
+          height: 90,
+          position: "absolute",
+          top: 95,
+          right: 0,
+        }}
+        onClose={() => {
+          dispatch({ type: "fetch error cleared" });
```
Comment (vincent-czi) [inline src/components/header/fetchError.tsx:21]: Use the `ACTION_TYPES.FETCH_ERROR_CLEAR` here rather than a bare string

**Review by vincent-czi — APPROVED** [review]
Overall good! Got some minor comments that would be good to address, but nothing I would need to look over again.

Related to PR, but different: we should also make a similar banner to indicate you're on the Staging version of the site. I can provide a Vite env var that can be checked to see if site is deployed as Staging version, then we can have a banner that appears based on that env var. If you want to make the banner but just not have it used anywhere right now, I can do the env var hook up later.

File: src/routes/fetchTree.tsx (hunk 1 above)
Comment (sidneymbell) [reply src/routes/fetchTree.tsx:34]: Yeah I generally agree. My thinking here was largely that the intended audience for Galago is largely folks who aren't technical... like at all. I barely (read: mostly don't) even know what CORS is, so I'm kind of assuming that users won't either / I'm not sure pointing them in that direction would actually help. 

I'm going to split the difference -- leave the error as-is for now, and I'll include an FAQ entry about "I see JSON at the end of my URL why doesn't fetch work" and link it from the error once it exists.

File: src/reducers/global.ts (hunk 2 above)
Comment (sidneymbell) [reply src/reducers/global.ts:365]: TY! Will update

File: src/components/header/fetchError.tsx (hunk 3 above)
Comment (sidneymbell) [reply src/components/header/fetchError.tsx:21]: ah ty

**Review by Janeece — APPROVED** [review]
LGTM! love the  color update, so much easier to read!

File: src/routes/fetchTree.tsx (hunk 1 above)
Comment (vincent-czi) [reply src/routes/fetchTree.tsx:34]: Yeah, it's also quite possible that the best course of action is for someone to think our site is broken and reach out to us. I don't know if we'll see that many users clamoring for help, and we would have the technical background to resolve it (hopefully) even if it's not technically _our_ bug.

**Comment by sidneymbell** [comment]
@vincent-czi -- good idea! I've added a `<StagingBanner/>` component and commented out the ternary in `header` that will replace `<BetaBanner/>` with the `<StagingBanner/>` once you've wired up the env variable to state.

**Pull request merged**